### PR TITLE
Adds PR template

### PR DESCRIPTION
**Pull request by amlcurran**
## This PR adds

A PR template so we can iterate and use it on Novoda projects

### Considerations and implementation

We have many different versions of PR templates, and this will allow us to have a single source of truth, whilst iterating and discussing. 

### Test(s) added 

Nope, this is Markdown

### Screenshots

No screenshots, but this is written using the new template. Meta.

### Paired with 

no-one but this has been discussed and iterated through a few projects

### CODE REVIEW

**Comment by mr-archano** [comment]
@charroch will love this
> No screenshots, but this is written using the new template. Meta.

**Comment by lgvalle** [comment]
Why `####` and not `#` ? :)

**Comment by lgvalle** [comment]
Consider that if we need to add sub-elements, an initial section block with size `####` is too small

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#### Test(s) added 
+
+ _what tests did you add? If none, explain why_ 
+
+#### Screenshots
+
+| Before | Spec  | After |
+| ------ | ----- | ----- |
+| gif/png _before_ | gif/png _spec_ | _gif/png after_ |
```
Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: This difference that I hated at other projects. 😄 
before and spec are italic: like `_before_`
but at the end, italic covers the whole `_gif/png after_`

Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: I have fixed that in our project's template (same as titles hierarchy :D)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+_Here you can talk more about the technical details or any interesting points_
+
+#### Test(s) added 
+
+ _what tests did you add? If none, explain why_ 
+
+#### Screenshots
+
+| Before | Spec  | After |
+| ------ | ----- | ----- |
```
Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can you delete 1 dash from the `Spec` underliner. 😄 It has 1 extra. 🙏 Sorry cannot help noticing

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#### This PR adds
```
Comment (mr-archano) [inline .github/PULL_REQUEST_TEMPLATE.md]: What about promote this to more prominent header (`###` vs `####`)? Same could be done for `Considerations and implementation`, while the other sub-sections could be considered accessory to the implementation, so it would be ok to leave them as they are.

**Comment by charroch** [comment]
@mr-archano did you see the news PR on the lib? They follow our template. This has huge value in terms of quality assessment.

Can we add LINT/Coverage or similar to the template? As in improved coverage by X

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+_Explain the value of the PR in a sentence._
+
+#### Considerations and implementation
+
+_Here you can talk more about the technical details or any interesting points_
+
+#### Test(s) added 
+
+ _what tests did you add? If none, explain why_
```
Comment (ataulm) [inline .github/PULL_REQUEST_TEMPLATE.md]: `why did you add tests around the areas you did? If none, explain why`?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: I will have a look.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: I like having `##` for the main sections (the first level is the PR title so that accounts for it) and using hierarchy (`###`, `####`) for sub-sections, instead of everything at the same level.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: Probably want to remove the heading space and capitalise the `w` in `what` for consistency

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: Not sure how helpful is having the spec tbh

**Comment by ataulm** [comment]
>Can we add LINT/Coverage or similar to the template? As in improved coverage by X

@charroch it's difficult since we'd need the old stats too - that's how sonar can do it. I would argue against this since it adds a lot of overhead to opening prs

**Comment by rock3r** [comment]
@charroch wouldn't that be usually calculated by PRB _after_ the PR is opened?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mr-archano) [inline .github/PULL_REQUEST_TEMPLATE.md]: It's a base template, and as such can be tweaked when included in another project: let's aim for something that catches a lot of the scenarios we're facing.

**Comment by mr-archano** [comment]
> @mr-archano did you see the news PR on the lib? They follow our template. This has huge value in terms of quality assessment.

@charroch I'm not sure I understand what you're referring to. Let's move this discussion to slack.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (mr-archano) [inline .github/PULL_REQUEST_TEMPLATE.md]: I found invaluable to have the JIRA link at the top of the description too. Should we add it @amlcurran?

Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: Basically, I wanted to make all of the titles the same size - because I wanted to make it clear that each section is important and none are subordinate. What do you think?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm of the mind to remove the spec from the default - and only add it on projects that we need it?

Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: Seems a good idea, especially if you end up adding a link to the ticket, which will contain the spec and all

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (rock3r) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think there is some hierarchy, especially if you think of it in a "we had a problem/task - here is the solution" optic:

image

(ignore the actual text in the sections, that's semi-random and could be improved)

Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: updated big titles https://github.com/novoda/novoda/pull/39/commits/ae340f1b3ed53a22f538cf491b1b46d79e24aa04 and JIRA link https://github.com/novoda/novoda/pull/39/commits/22d77938500cf9c1d2da522231f586a88b317881

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: I personally think this is OK - saying "what" tests are added kinda of forces you to say what they're doing?

Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: https://github.com/novoda/novoda/pull/39/commits/2bdeb377fdb6ff9c1ddc388339484d808399d903 not sure what you mean by heading space?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: Link to a JIRA ticket helps in that case. Specs are usually 1 click away.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: I agree with you on this @amlcurran 
I noticed before, just because `Tests` section is smaller, it made people think it is less important.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+_Explain the value of the PR in a sentence._
+
+## Considerations and implementation
+
+_Here you can talk more about the technical details or any interesting points_
+
+### Test(s) added 
+
+ _What tests did you add? If none, explain why_
```
Comment (ataulm) [inline .github/PULL_REQUEST_TEMPLATE.md]: https://github.com/novoda/novoda/pull/39#discussion_r98191784

I don't think it forces you at all - I gave the example on the other PR you linked, where it ended up people just indexed the tests they added, which is not useful.

Comment (tasomaniac) [inline .github/PULL_REQUEST_TEMPLATE.md]: I agree with @ataulm on this. 
The answer to the question: `What tests did you add?` is already available in the PR itself. In `Files Changed`.
The question should be more like: `Which parts of the implementation did you cover with tests?`

Comment (ataulm) [inline .github/PULL_REQUEST_TEMPLATE.md]: also works for me, any sort of prompt to give a more deeper answer

Comment (amlcurran) [inline .github/PULL_REQUEST_TEMPLATE.md]: OK, I'd like to be more around functionality (e.g. "Added tests around displaying a notification" rather than "Tested NotificationPresenter.java") but maybe we can improve that/I can submit a second PR. Will update to ataul's initial suggestion.

**Comment by ataulm** [comment]
I think carl wanted paul and volker to have a look before merging @lgvalle

**Comment by blundell** [comment]
ya fail @lgvalle

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+_Explain the value of the PR in a sentence._
+
+## Considerations and implementation
+
+_Here you can talk more about the technical details or any interesting points_
+
+### Test(s) added 
+
+_Why did you add tests around the areas you did? If none, explain why_
```
Comment (blundell) [inline .github/PULL_REQUEST_TEMPLATE.md]: 👍
